docs/mosaicking.rst

```diff
@@ -0,0 +1,310 @@
+but this may not always be optimal because if the mosaic is rotated compared to
+North, there may be a lot of empty space in the final mosaic. The ``auto_rotate``
+option can therefore be used to find the optimal rotation for the WCS that minimizes
+the area of the final mosaic file:
+
+.. doctest-requires:: shapely
+
+    >>> wcs_out, shape_out = find_optimal_celestial_wcs(m17_hdus,
+    ...                                                 frame=Galactic(),
+    ...                                                 auto_rotate=True)
+
+Note that this requires `Shapely <...>`_ 1.6 or later to be installed. We can now
+look at the final WCS and shape:
+
+.. doctest-requires:: shapely
+
+    >>> wcs_out.to_header()  # doctest: +FLOAT_CMP
+    WCSAXES =                    2 / Number of coordinate axes
+    CRPIX1  =      1102.3949574309 / Pixel coordinate of reference point
+    CRPIX2  =      900.46211361965 / Pixel coordinate of reference point
+    PC1_1   =     0.88188439264557 / Coordinate transformation matrix element
+    PC1_2   =     0.47146571244169 / Coordinate transformation matrix element
+    PC2_1   =    -0.47146571244169 / Coordinate transformation matrix element
+    PC2_2   =     0.88188439264557 / Coordinate transformation matrix element
+    CDELT1  =     -0.0002777777845 / [deg] Coordinate increment at reference point
+    CDELT2  =      0.0002777777845 / [deg] Coordinate increment at reference point
+    CUNIT1  = 'deg'                / Units of coordinate increment and value
+    CUNIT2  = 'deg'                / Units of coordinate increment and value
+    CTYPE1  = 'GLON-TAN'           / galactic longitude, gnomonic projection
+    CTYPE2  = 'GLAT-TAN'           / galactic latitude, gnomonic projection
+    CRVAL1  =      15.116960053834 / [deg] Coordinate value at reference point
+    CRVAL2  =    -0.72166403860488 / [deg] Coordinate value at reference point
+    LONPOLE =                180.0 / [deg] Native longitude of celestial pole
+    LATPOLE =    -0.72166403860488 / [deg] Native latitude of celestial pole
+    >>> shape_out
+    (1800, 2201)
+
+As expected, the optimal shape is smaller than was returned previously.
+
+Resolution
+----------
+
+By default, the final mosaic will have the resolution of the highest resolution
+input image, but this can be overriden using the ``resolution=`` keyword
+argument::
+
+    >>> from astropy import units as u
+    >>> wcs_out, shape_out = find_optimal_celestial_wcs(m17_hdus,
+    ...                                                 resolution=1.5 * u.arcsec)
+
+Projection and reference coordinate
+-----------------------------------
+
+Finally, you can customize the projection to use as well as the reference
+coordinate. To change the projection from the default (which is the
+gnomonic projection, or ``TAN``), you can use the ``projection=`` keyword
+argument, which should be set to a valid three-letter FITS-WCS projection code::
```

Comment:
link this to http://adsabs.harvard.edu/abs/2002A%26A...395.1061G?

Reply:
Done